### PR TITLE
Team1

### PR DESCRIPTION
**Pull request by Fadi-Rasheed**

### CODE REVIEW

File: api/src/middlewares/submit-assignment.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+const express = require('express');
```
Comment (alxbrg) [inline api/src/middlewares/submit-assignment.js]: I don't understand the use of `express` here. You're only using it to serve static files, which is not part of the assignment nor the specs we agreed upon for the API.

File: api/src/apply.js [hunk 2]
```diff
@@ -48,13 +48,29 @@ const sendEmail = (toEmail, Data, Subject) => {
...
-        fromEmail,
-        applyToOrgTemplate({ params: req.body }),
-        'A new student applied'
-    ).then(() => {
+
+    });
+}
+
+module.exports = (req, res) => {
+    createApplicant(req.body).then(() => {
```
Comment (alxbrg) [inline api/src/apply.js]: What you did in your `createApplicant` function makes sense (well taking into account my comment above). But this function will actually be implemented by another team. Like I said on Tuesday, you should use a placeholder for now. Something like: 
```js
const createApplicatant = async (applicant) => {}
``` 
You can still check the `body` though to make sure that the correct fields are there.

File: api/src/middlewares/submit-assignment.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+//creat a folder to store the photos 
+app.use(express.static("./uploads"));
+
+app.use(express.json());
+
+//destination of the storage
+const destination = './uploads/';  
+
+const storage = multer.diskStorage({
```
Comment (alxbrg) [inline api/src/middlewares/submit-assignment.js]: Another team is supposed to take care of the storage. We don't even know how they will decide to do it. `multer` offers the possibility to delay the storage using an in-memory buffer. We can talk about that tomorrow.

File: api/src/middlewares/submit-assignment.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    const allowedFileTypes = /jpeg|jpg|png|gif/;
+    const mimetype = allowedFileTypes.test(file.mimetype);
+    if (mimetype) {
+        return cb(null,true)
+    } else{
+        cb("Error : images only")
+    }
+}
+
+let newAssignment
```
Comment (alxbrg) [inline api/src/middlewares/submit-assignment.js]: I don't understand this. What's the intention of using this variable here?

File: api/src/middlewares/submit-assignment.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+module.exports =  (req, res) => {
+        upload(req,res,(err) => {
```
Comment (alxbrg) [inline api/src/middlewares/submit-assignment.js]: Makes sense in the context of the rest of your code, but we have to call a function written by another team here. One that stores both some fields and the files. The definition for this function can be found in my own PR. 
Also, one thing that's missing: we should send emails to both the applicant and HYF here as well.

File: api/src/middlewares/submit-assignment.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+const upload = multer({
+    storage: storage,
+    limits:{fileSize: 1024*1024*5},
+    fileFilter: (req,file,cb)=>{
+        checkFileType(file,cb);
+    }
+    }).single('myImage'); // here should be entered the name of the input field of the file 
+
+    function checkFileType (file,cb){
```
Comment (alxbrg) [inline api/src/middlewares/submit-assignment.js]: Great function! But you should define that in the global scope, not inside of another function. Otherwise, this function gets defined and then destroyed on each call of the `upload` function.

File: api/src/middlewares/submit-assignment.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+let newAssignment
+
+module.exports =  (req, res) => {
+        upload(req,res,(err) => {
+            if(err){
+                res.send(err);
+            } else {
+                res.send('Your Assignment has been uploaded');
```
Comment (alxbrg) [inline api/src/middlewares/submit-assignment.js]: If you look at the existing `apply` middle-ware, you'll see that the response should have this format:
```
res.status(200).json({ message: 'You got an email :-)'});
```
We need to stay consistent here.

**Comment by M3kH** [comment]
Is this still valid or we can burn it 🔥with fire?

**Comment by Fadi-Rasheed** [comment]
we should burn it.
